### PR TITLE
Add doc readme

### PR DESCRIPTION
**Pull request by yiyixuxu**
1. added a doc README page
2. added a "Schedulers" section to the doc similar to "Pipelines": an "overview" page and separate pages for each scheduler
3. fixed the links in Pipelines overview page

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -166,3 +166,6 @@ tags
 .DS_Store
 # RL pipelines may produce mp4 outputs
 *.mp4
+
+# dependencies
+/transformers
```
Comment (pcuenca) [inline .gitignore:171]: Is this necessary?

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+```
+Sections that were moved:
+
+[ <a href="../new-file#section-b">Section A</a><a id="section-a"></a> ]
+```
+
+Use the relative style to link to the new file so that the versioned docs continue to work.
+
+For an example of a rich moved section set please see the very end of [the Trainer doc](https://github.com/huggingface/transformers/blob/main/docs/source/en/main_classes/trainer.mdx).
```
Comment (pcuenca) [inline docs/README.md:112]:
```suggestion
For an example of a rich moved section set please see the very end of [the transformers Trainer doc](https://github.com/huggingface/transformers/blob/main/docs/source/en/main_classes/trainer.mdx).
```

Just to clarify this is not a diffusers trainer.

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+# Generating the documentation
+
+To generate the documentation, you first have to build it. Several packages are necessary to build the doc, 
+you can install them with the following command, at the root of the code repository:
+
+```bash
+pip install -e ".[docs]"
+```
+
+Then you need to install our special tool that builds the documentation:
```
Comment (pcuenca) [inline docs/README.md:26]:
```suggestion
Then you need to install our open source documentation builder tool:
```

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+Sections that were moved:
+
+[ <a href="#section-b">Section A</a><a id="section-a"></a> ]
+```
+and of course, if you moved it to another file, then:
+
+```
+Sections that were moved:
+
+[ <a href="../new-file#section-b">Section A</a><a id="section-a"></a> ]
```
Comment (pcuenca) [inline docs/README.md:107]: Very cool, I didn't know this was possible! 🙏

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+The `huggingface/diffusers` documentation follows the
+[Google documentation](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) style for docstrings,
+although we can write them directly in Markdown.
+
+### Adding a new tutorial
+
+Adding a new tutorial or section is done in two steps:
+
+- Add a new file under `./source`. This file can either be ReStructuredText (.rst) or Markdown (.md).
```
Comment (pcuenca) [inline docs/README.md:125]: Maybe `docs/source`? Or clarify in a previous sentence that these paths refer to the root of the documentation at `docs`.

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+
+Make sure to put your new file under the proper section. It's unlikely to go in the first section (*Get Started*), so
+depending on the intended targets (beginners, more advanced users, or researchers) it should go in sections two, three, or four.
+
+### Adding a new pipeline/scheduler
+
+When adding a new pipeline:
+
+- create a file `xxx.mdx` under `./source/api/pipelines` (don't hesitate to copy an existing file as template).
+- Link that file in (*Diffusers Summary*) section in `./source/api/pipelines/overview.mdx`, along with the link to the paper, and a colab notebook (if available).
```
Comment (pcuenca) [inline docs/README.md:136]:
```suggestion
- Link that file in the (*Diffusers Summary*) section in `./source/api/pipelines/overview.mdx`, along with the link to the paper, and a colab notebook (if available).
```

File: docs/README.md [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+          Total loss is the sum of the masked language modeling loss and the next sequence prediction (classification) loss.
+        - **prediction_scores** (`torch.FloatTensor` of shape `(batch_size, sequence_length, config.vocab_size)`) --
+          Prediction scores of the language modeling head (scores for each vocabulary token before SoftMax).
+```
+
+#### Adding an image
+
+Due to the rapidly growing repository, it is important to make sure that no files that would significantly weigh down the repository are added. This includes images, videos, and other non-text files. We prefer to leverage a hf.co hosted `dataset` like
+the ones hosted on [`hf-internal-testing`](https://huggingface.co/hf-internal-testing) in which to place these files and reference
+them by URL. We recommend putting them in the following dataset: [huggingface/documentation-images](https://huggingface.co/datasets/huggingface/documentation-images).
```
Comment (pcuenca) [inline docs/README.md:265]:
```suggestion
Due to the rapidly growing repository, it is important to make sure that no files that would significantly weigh down the repository are added. This includes images, videos, and other non-text files. We prefer to leverage a hf.co hosted `dataset` to place these files and reference
them by URL. We recommend putting them in the following dataset: [huggingface/documentation-images](https://huggingface.co/datasets/huggingface/documentation-images).
```

Why mention `hf-internal-testing` if we recommend `documentation-images`? `hf-internal-testing` should be useful for tests, but that could be covered in another doc :)

File: docs/README.md [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+If an external contribution, feel free to add the images to your PR and ask a Hugging Face member to migrate your images
+to this dataset.
+
+## Styling the docstring
+
+We have an automatic script running with the `make style` comment that will make sure that:
```
Comment (pcuenca) [inline docs/README.md:271]:
```suggestion
We have an automatic script running with the `make style` command that will make sure that:
```

File: docs/source/api/schedulers/overview.mdx [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+- Schedulers define the methodology for iteratively adding noise to an image or for updating a sample based on model outputs.
+    - adding noise in different manners represent the algorithmic processes to train a diffusion model by adding noise to images.
+    - for inference, the scheduler defines how to update a sample based on an output from a pretrained model.
+- Schedulers are often defined by a *noise schedule* and an *update rule* to solve the differential equation solution.
+
+### Discrete versus continuous schedulers
+
+All schedulers take in a timestep to predict the updated version of the sample being diffused.
+The timesteps dictate where in the diffusion process the step is, where data is generated by iterating forward in time and inference is executed by propagating backwards through timesteps.
+Different algorithms use timesteps that both discrete (accepting `int` inputs), such as the [`DDPMScheduler`] or [`PNDMScheduler`], and continuous (accepting `float` inputs), such as the score-based schedulers [`ScoreSdeVeScheduler`] or [`ScoreSdeVpScheduler`].
```
Comment (pcuenca) [inline docs/source/api/schedulers/overview.mdx:30]:
```suggestion
Different algorithms use timesteps that can be discrete (accepting `int` inputs), such as the [`DDPMScheduler`] or [`PNDMScheduler`], or continuous (accepting `float` inputs), such as the score-based schedulers [`ScoreSdeVeScheduler`] or [`ScoreSdeVpScheduler`].
```

File: docs/source/api/schedulers/overview.mdx [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+## Designing Re-usable schedulers
+
+The core design principle between the schedule functions is to be model, system, and framework independent.
+This allows for rapid experimentation and cleaner abstractions in the code, where the model prediction is separated from the sample update.
+To this end, the design of schedulers is such that:
+
+- Schedulers can be used interchangeably between diffusion models in inference to find the preferred trade-off between speed and generation quality.
+- Schedulers are currently by default in PyTorch, but are designed to be framework independent (partial Jax support currently exists).
```
Comment (pcuenca) [inline docs/source/api/schedulers/overview.mdx:39]:
```suggestion
- Schedulers are currently available in PyTorch by default, but are designed to be framework independent (partial Jax support currently exists).
```

File: docs/source/api/schedulers/overview.mdx [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+| [ipndm](./ipndm) | [**improved pseudo numerical methods for diffusion models (iPNDM)**](https://github.com/crowsonkb/v-diffusion-pytorch/blob/987f8985e38208345c1959b0ea767a625831cc9b/diffusion/sampling.py#L296) |
+| [score_sde_vp](./score_sde_vp) | [**Variance preserving stochastic differential equation (VP-SDE) scheduler**](https://arxiv.org/abs/2011.13456) |
+| [euler](./euler) | [**Euler scheduler**](https://arxiv.org/abs/2206.00364) |
+| [euler_ancestral](./euler_ancestral) | [**Euler Ancestral scheduler**](https://github.com/crowsonkb/k-diffusion/blob/481677d114f6ea445aa009cf5bd7a9cdee909e47/k_diffusion/sampling.py#L72) |
+| [vq_diffusion](./vq_diffusion) | [**VQDiffusionScheduler**](https://arxiv.org/abs/2111.14822) |
+| [repaint](./repaint) | [**RePaint scheduler**](https://arxiv.org/abs/2201.09865) |
+
+## API
+
+The core API for any new scheduler must follow a limited structure.
```
Comment (pcuenca) [inline docs/source/api/schedulers/overview.mdx:68]:
```suggestion
The core API for any new scheduler must follow a specific structure.
```

**Review by pcuenca — APPROVED** [review]
Amazing improvements, great work!

I left a few suggestions that are just to be considered that, feel free to ignore if you don't think they should be applied :)

File: docs/README.md [hunk 12]
```diff
@@ -0,0 +1,278 @@
+<!---
+Copyright 2020 The HuggingFace Team. All rights reserved.
```
Comment (patrickvonplaten) [inline docs/README.md:2]:
```suggestion
Copyright 2022- The HuggingFace Team. All rights reserved.
```

File: docs/README.md [hunk 13]
```diff
@@ -0,0 +1,278 @@
...
+you can install them with the following command, at the root of the code repository:
+
+```bash
+pip install -e ".[docs]"
+```
+
+Then you need to install our special tool that builds the documentation:
+
+```bash
+pip install git+https://github.com/huggingface/doc-builder
```
Comment (patrickvonplaten) [inline docs/README.md:29]:
```suggestion
pip install hf-doc-builder
```

File: docs/README.md [hunk 14]
```diff
@@ -0,0 +1,278 @@
...
+
+---
+**NOTE**
+
+You only need to generate the documentation to inspect it locally (if you're planning changes and want to
+check how they look before committing for instance). You don't have to commit the built documentation.
+
+---
+
+## Building the documentation
```
Comment (patrickvonplaten) [inline docs/README.md:40]: Actually I don't think that this section is really necessary when writing docs

@mishig25 what do you think?

Comment (patrickvonplaten) [reply docs/README.md:40]: Preview should be enough no?

File: docs/README.md [hunk 15]
```diff
@@ -0,0 +1,278 @@
...
+
+```bash
+doc-builder build diffusers docs/source/ --build_dir ~/tmp/test-build
+```
+
+You can adapt the `--build_dir` to set any temporary folder that you prefer. This command will create it and generate
+the MDX files that will be rendered as the documentation on the main website. You can inspect them in your favorite
+Markdown editor.
+
+## Previewing the documentation
```
Comment (patrickvonplaten) [inline docs/README.md:53]: That's the important section :-)

File: docs/README.md [hunk 16]
```diff
@@ -0,0 +1,278 @@
...
+doc-builder preview {package_name} {path_to_docs}
+```
+
+For example:
+
+```bash
+doc-builder preview diffusers docs/source/
+```
+
+The docs will be viewable at [http://localhost:3000](http://localhost:3000). You can also preview the docs once you have opened a PR. You will see a bot add a comment to a link where the documentation with your changes lives.
```
Comment (patrickvonplaten) [inline docs/README.md:73]: nice!

File: docs/README.md [hunk 17]
```diff
@@ -0,0 +1,278 @@
...
+### Adding a new pipeline/scheduler
+
+When adding a new pipeline:
+
+- create a file `xxx.mdx` under `./source/api/pipelines` (don't hesitate to copy an existing file as template).
+- Link that file in (*Diffusers Summary*) section in `./source/api/pipelines/overview.mdx`, along with the link to the paper, and a colab notebook (if available).
+- Write a short overview of the diffusion model:
+    - Overview with paper & authors
+    - Paper abstract
+    - Tips and tricks and how to use it best
```
Comment (patrickvonplaten) [inline docs/README.md:140]:
```suggestion
    - Tips and tricks and how to use it best
    - Possible an end-to-end example of how to use it
```

File: docs/README.md [hunk 18]
```diff
@@ -0,0 +1,278 @@
...
+## Styling the docstring
+
+We have an automatic script running with the `make style` comment that will make sure that:
+- the docstrings fully take advantage of the line width
+- all code examples are formatted using black, like the code of the Transformers library
+
+This script may have some weird failures if you made a syntax mistake or if you uncover a bug. Therefore, it's
+recommended to commit your changes before running `make style`, so you can revert the changes done by that script
+easily.
+
```
Comment (patrickvonplaten) [inline docs/README.md:278]: Very nice!

File: docs/source/api/pipelines/overview.mdx [hunk 19]
```diff
@@ -44,31 +44,31 @@ available a colab notebook to directly try them out.
 
 | Pipeline | Paper | Tasks | Colab
 |---|---|:---:|:---:|
-| [alt_diffusion](./api/pipelines/alt_diffusion) | [**AltDiffusion**](https://arxiv.org/abs/2211.06679) | Image-to-Image Text-Guided Generation | -
```
Comment (patrickvonplaten) [inline docs/source/api/pipelines/overview.mdx:47]: thanks!

File: docs/source/api/schedulers.mdx [hunk 20]
```diff
@@ -56,128 +56,3 @@ The class [`SchedulerOutput`] contains the outputs from any schedulers `step(...
 
 [[autodoc]] schedulers.scheduling_utils.SchedulerOutput
```
Comment (patrickvonplaten) [inline docs/source/api/schedulers.mdx:57]: Let's maybe fully remove this doc page and move everything which is still left here to "overview.mdx"

File: .gitignore (hunk 1 above)
Comment (patrickvonplaten) [reply .gitignore:171]: Yes think we don't need this, but also ok for me to leave it there

File: docs/README.md (hunk 14 above)
Comment (yiyixuxu) [reply docs/README.md:40]: this section actually really really confused me lol- we should definitely remove it if not needed, or make it clear when it is needed

File: docs/source/api/schedulers.mdx (hunk 20 above)
Comment (yiyixuxu) [reply docs/source/api/schedulers.mdx:57]: > Also, maybe we can add a brief discussion about schedulers working with different objectives (epsilon, sample, or v-prediction), but that can come in another PR.

good idea! I can add this to the overview page
